src/ui/LiveSplit.tsx

```diff
@@ -245,6 +245,7 @@ export class LiveSplit extends React.Component<{}, State> {
     }
 
     public saveSplits() {
+        this.state.timer.writeWith((t) => t.reset(true));
         const lss = this.state.timer.readWith((t) => t.getRun().saveAsLss());
```

Comment:
Also as an additional hint, you are locking the timer twice here (writeWith, readWith). This is fine, since they are separate, so it's not a deadlock, however for the very short period of time in between, something else could lock the timer and completely modify its state, meaning that it may not actually be properly reset anymore. So you actually want to combine these into just a single writeWith.

Reply:
I will keep this in mind when I get around to making this match livesplit's behavior